### ISSUE TITLE
"If block ends with a return ... " in chain of else ifs

### ISSUE THREAD

**Issue by 200sc** [golang/lint#290]
In this code:

``` Go
func golintTest() {
	r := 0
	if r == 0 {
		r = 5
	} else if r == 3 {
		r = 2
	} else if r == 2 {
		r = -2
		return
	} else {
		panic("something invalid")
	}
        fmt.Println(r)
}

```

(Yes the actions going on are meaningless)

In this case, golint will suggest that we drop the else as the last 'else if' case returns. 
That seems incorrect, if an earlier branch was taken.

Interestingly this will not happen if there is a single else if case with a return, and it
also won't happen if we add a third else if case before the return case.

**Comment by 200sc** [comment]
Sorry, looks like this has been issued before and is in progress? #181
